### PR TITLE
Show tab if user promted to save it

### PR DESCRIPTION
**Pull request by stereobooster**
Companion PR for https://github.com/atom/atom/pull/12926

### CODE REVIEW

**Comment by thomasjo** [comment]
@50Wliu Care to elaborate on what changes are required?

Personally, I think we should add at least one test to verify this functionality. Tests are vital to guarding against future regressions.

**Comment by Ben3eeE** [comment]
@thomasjo this change seems to break some specs and I have added comments in the atom/atom PR. This is a work in progress.

**Comment by stereobooster** [comment]
My change makes `closeTab` asynchronous. But it is assumed to be synchronous e.g.

this fails

```coffeescript
          tabBar.getTabs()[0].querySelector('.close-icon').click()
          tabBar.getTabs()[1].querySelector('.close-icon').click()
          window.timeouts.map ([id, time, callback]) => callback()
```

but this not

```coffeescript
          tabBar.getTabs()[0].querySelector('.close-icon').click()
          tabBar.getTabs()[1].querySelector('.close-icon').click()
          window.timeouts.map ([id, time, callback]) => callback()
```

File: lib/tab-bar-view.coffee [hunk 1]
```diff
@@ -162,8 +162,16 @@ class TabBarView extends HTMLElement
...
-    @pane.destroyItem(tab.item) if tab?
 
+    return unless tab?
+
+    if tab.item.isModified?()
+      @setActiveTab(tab)
+      @pane.activateItem(tab.item)
+
+    callback = => @pane.destroyItem(tab.item)
+    setTimeout(callback, 100)
```
Comment (stereobooster) [inline lib/tab-bar-view.coffee]: Tried  `setImmediate` but it worked 50% of time

File: spec/tabs-spec.coffee [hunk 2]
```diff
@@ -16,10 +16,16 @@ addItemToPane = (pane, item, index) ->
   else
     throw new Error("Unspoorted pane.addItem API")
 
+waitsForTimeout = ->
+  window.timeouts.map ([id, time, callback]) -> callback()
```
Comment (stereobooster) [inline spec/tabs-spec.coffee]: I need some response on this from core

File: spec/tabs-spec.coffee [hunk 3]
```diff
@@ -619,21 +633,26 @@ describe "TabBarView", ->
...
         expect(pane.getItems().length).toBe 2
         expect(pane.getItems().indexOf(item2)).toBe -1
         expect(tabBar.getTabs().length).toBe 2
         expect($(tabBar).find('.tab:contains(Item 2)')).not.toExist()
 
       it "does nothing if no tabs are open", ->
         atom.commands.dispatch(paneElement, 'tabs:close-tab')
+        waitsForTimeout()
         atom.commands.dispatch(paneElement, 'tabs:close-tab')
+        waitsForTimeout()
```
Comment (stereobooster) [inline spec/tabs-spec.coffee]: change the way this code works e.g. this works other way:
```
          atom.commands.dispatch(paneElement, 'tabs:close-tab')
          atom.commands.dispatch(paneElement, 'tabs:close-tab')
         waitsForTimeout()
```

**Comment by stereobooster** [comment]
@Ben3eeE can you take a look or cc someone. Thanks

**Pull request closed without merging**